g2p/mappings/utils.py

```diff
@@ -661,7 +661,13 @@ class _MappingModelDefinition(BaseModel):
     """Deprecated: Please use rule_ordering='as_written' """
 
     case_sensitive: bool = True
```

Comment:
I would like to see some tests about interactions between `case_sensitive` and `preserve_case`. Does `case_sensitive=False` and `preserve_case=True` work for example?

Reply:
Hum, good question. It actually only makes sense to have `case_sensitive=False` when you set `preserve_case=True`. Having both be true should raise an error.

Reply:
The semantics of preserve case is that case does affect the meaning of the characters (i.e., the mapping is case insensitive) but should be preserved because it has non-phonetic meaning (e.g., proper name, beginning of the sentence).

Reply:
So I just added a model validator that raises when the two are set incompatibly.